geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterElderManager.java

```diff
@@ -84,11 +84,10 @@ public boolean isElder() {
 
   public ElderState getElderState(boolean waitToBecomeElder) throws InterruptedException {
     if (waitToBecomeElder) {
-      // This should always return true.
       waitForElder(clusterDistributionManager.getId());
     }
 
-    if (!isElder()) {
+    if (!isElder() && !waitToBecomeElder) {
```

Comment:
If this member is not elder and is not waiting to become elder (which I assume means this member wants to become elder), then should we still initialize the elder state on this member as we do on line 94?  I'm not sure of the side effects, but logically it seems like this member is becoming elder even if it "didn't intend to" via the watToBecomeElder flag.

Reply:
If waitToBecomeElder is true, it means we expect this member to become the elder based on the logic how the elder is chosen in the cluster. For only way that it does not wait is when this member is being shutdown. In this case the cluster will have a new DLockGrantor after choose a new Elder (via membership listener). 

This fix just follows the old implementation to avoid the NPE. Please see the analysis in the JIRA ticket.